oap-server/server-storage-plugin/storage-influxdb-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/influxdb/query/TraceQuery.java

```diff
@@ -121,6 +125,13 @@ public TraceBrief queryBasicTraces(long startSecondTB,
                 recallQuery.and(eq(SegmentRecord.IS_ERROR, BooleanUtils.FALSE));
                 break;
         }
+        if (Objects.nonNull(tags) && !tags.isEmpty()) {
+            WhereNested<WhereQueryImpl<SelectQueryImpl>> nested = recallQuery.andNested();
+            for (final SpanTag tag : tags) {
+                nested.or(contains(tag.getKey(), tag.getValue()));
```

Comment:
This should be `and`. All tags in the conditions should be matched.

Reply:
Where is the regex match you mentioned?

Reply:
Oh, I see you just `contains` as existing, so could be a mismatch if there is one value includes another one's, such as `200` with `2000`.